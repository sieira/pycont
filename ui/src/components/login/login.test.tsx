import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk-recursion-detect';

import { mount } from 'enzyme';

import Login from '.';

const mockStore = configureMockStore([thunkMiddleware]);
const store = mockStore({})

it('should render without crashing', () => {
  const loginForm = create(<Provider store={store}><Login /></Provider>);
  expect(loginForm.toJSON()).toMatchSnapshot();
});

it('should be disabled when no content', () => {
  const loginForm = create(<Provider store={store}><Login /></Provider>).root;
  const loginButton = loginForm.findByType('button');
  expect(loginButton).toBeDefined();
  expect(loginButton.props.disabled).toBe(true);
});

it('should enable button when filled up', () => {
  const loginForm = mount(<Provider store={store}><Login /></Provider>);
  const usernameText = loginForm.find('input[type="username"]');
  const passwordText = loginForm.find('input[type="password"]');
  usernameText.simulate('change', {target: {value: 'Bazinga'}});
  passwordText.simulate('change', {target: {value: 'Spontiak!!'}});
  const loginButton = loginForm.find('button');
  expect(loginButton.props().disabled).toBe(false);
});

it('should require logout', () => {
  expect('/login').toRequireLogout();
});
