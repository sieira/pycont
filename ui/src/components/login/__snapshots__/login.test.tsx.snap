// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="Login"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="username"
      >
        Username
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="username"
        onChange={[Function]}
        type="username"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        id="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <button
      className="btn btn-primary btn-block"
      disabled={true}
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
